design-toolkit/components/src/components/notice/index.tsx

```diff
@@ -0,0 +1,316 @@
+    NoticeQueueEvent | NoticeDequeueEvent | NoticeActionEvent
+  >();
+  const emitActionPrimary = useEmit(NoticeEventTypes.actionPrimary);
+  const emitActionSecondary = useEmit(NoticeEventTypes.actionSecondary);
+  const emitClose = useEmit(NoticeEventTypes.close);
+
+  useOn(NoticeEventTypes.queue, (data) => {
+    if ((id && data.payload.target === id) || !id) {
+      queue.add(
+        {
+          ...data.payload,
+          id: data.payload.id || uuid(),
+          color: defaultColor || data.payload.color,
+        },
+        {
+          timeout: defaultTimeout ?? data.payload.timeout,
+        },
+      );
+    }
+  });
+
+  useOn(NoticeEventTypes.dequeue, (data) => {
+    // @ts-expect-error : queue.queue exists, but is not currently documented
+    const queuedNotices = queue.queue;
+
+    const dequeue = queuedNotices.filter((toast: QueuedToast<NoticeContent>) =>
+      matchesMetadata(data.payload, toast.content),
+    );
+
+    if (dequeue.length && dequeue.length === queuedNotices.length) {
+      queue.clear();
+    } else {
+      for (const toast of dequeue) {
+        queue.close(toast.key);
+      }
+    }
+  });
+
+  useEffect(() => {
+    queue.subscribe(() => {
+      setHasNotices(queue.visibleToasts.length > 0);
+    });
+  });
+
+  const children = (
+    <>
+      {!hideClearAll && hasNotices && (
+        <Button
+          className={composeRenderProps(
+            classNames?.button,
+            (className) => className ?? '',
+          )}
+          variant='outline'
+          onPress={queue.clear}
+        >
+          Clear All
+        </Button>
+      )}
+      <ToastList
+        className={composeRenderProps(classNames?.list, (className) =>
+          list({ className }),
+        )}
+      >
+        {({ toast }: { toast: QueuedToast<NoticeContent> }) => (
+          <Notice
+            {...toast.content}
+            key={toast.key}
+            showClose={
+              !(
+                toast.timeout &&
+                toast.content.primary &&
+                toast.content.secondary
+              )
+            }
+            classNames={classNames?.notice}
+            size={size}
+            onPrimaryAction={() => emitActionPrimary({ id: toast.content.id })}
+            onSecondaryAction={() =>
+              emitActionSecondary({ id: toast.content.id })
+            }
+            onClose={() => {
+              toast.onClose?.();
+              queue.close(toast.key);
+              emitClose({ id: toast.content.id });
+            }}
+          />
+        )}
+      </ToastList>
+    </>
+  );
+
+  return global ? (
+    <ToastRegion
+      className={composeRenderProps(classNames?.region, (className) =>
+        region({ className }),
+      )}
+      data-placement={placement}
+      queue={queue}
+    >
+      {children}
+    </ToastRegion>
+  ) : (
+    <div
+      {...regionProps}
+      className={region({ className: classNames?.region as string })}
```

Comment:
You're going to need to either limit this className to a string (can' be passing a functional call to this prop) or creating a discrimination union type that states that when global is true, that className is able to be a function